src/emeus-constraint-layout.c

```diff
@@ -1407,7 +1407,7 @@ create_child_constraint (EmeusConstraintLayout      *layout,
   /* Alternatively, if it's not a constant value, we find the variable
    * associated with it
    */
-  if (constraint->source_object != NULL)
+  if (constraint->source_object != NULL && !EMEUS_IS_CONSTRAINT_LAYOUT (constraint->source_object))
```

Comment:
The check is not entirely enough.

We need to check:

 * the `source_object` must be `NULL` OR
 * the `source_object` is a `EmeusConstraintLayout` AND
 * the `source_object` is the same `layout`

Otherwise it's possible to pass a `source_object` as a separate `EmeusConstraintLayout`, which would be a mess because we wouldn't have access to its solver.